packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -74,123 +90,522 @@ const propTypes = {
    */
   required: PropTypes.bool,
   /**
-   * @private Internal prop for showing date picker.
+   * @private
+   * Internal prop for showing date picker.
    */
   shouldShowPicker: PropTypes.bool,
   /**
    * The selected or entered date value to display in the date input.
    */
   value: PropTypes.string,
-  /**
-  * String that labels the current element. 'aria-label' must be present,
-  * for accessibility.
-  */
-  ariaLabel: PropTypes.string,
 };
 
 const defaultProps = {
+  ariaLabel: undefined,
   buttonRefCallback: undefined,
   inputAttributes: undefined,
+  inputRefCallback: undefined,
   isIncomplete: false,
   isInvalid: false,
   name: undefined,
   onBlur: undefined,
+  onButtonFocus: undefined,
   onChange: undefined,
   onClick: undefined,
   onFocus: undefined,
-  onButtonFocus: undefined,
   onKeyDown: undefined,
   required: false,
   placeholder: undefined,
   value: undefined,
-  ariaLabel: undefined,
 };
 
-class DatePickerInput extends React.Component {
-  constructor(props) {
-    super(props);
+const dateReducer = (state, inputValue) => ({
+  ...state,
+  ...inputValue,
+});
+
+const dateState = { day: '', month: '', year: '' };
+
+const DatePickerInput = (props) => {
+  const {
+    ariaLabel,
+    buttonRefCallback,
+    inputAttributes,
+    inputRefCallback,
+    intl,
+    isIncomplete,
+    isInvalid,
+    name,
+    onBlur,
+    onButtonFocus,
+    onChange,
+    onClick,
+    onFocus,
+    onKeyDown,
+    placeholder,
+    required,
+    value,
+    ...customProps
+  } = props;
 
-    this.handleOnButtonClick = this.handleOnButtonClick.bind(this);
-    this.handleOnChange = this.handleOnChange.bind(this);
-    this.handleOnKeyDown = this.handleOnKeyDown.bind(this);
+  const [date, dateDispatch] = useReducer(dateReducer, dateState);
+  const [isFocused, setFocused] = useState(false);
+  const [dayInitialFocused, setDayInitialFocused] = useState(false);
+  const [monthInitialFocused, setMonthInitialFocused] = useState(false);
+  const [yearInitialFocused, setYearInitialFocused] = useState(false);
+  const dayInputRef = useRef();
+  const monthInputRef = useRef();
+  const yearInputRef = useRef();
+
+  const id = customProps.id ? customProps.id : undefined;
+  if (customProps.id) {
+    delete customProps.id;
   }
 
-  componentDidUpdate(prevProps) {
-    if (this.shouldShowPicker && !prevProps.shouldShowPicker && this.props.onClick) {
-      this.props.onClick();
+  const { onCalendarButtonClick, shouldShowPicker } = customProps;
+  delete customProps.onCalendarButtonClick;
+  delete customProps.shouldShowPicker;
+
+  const additionalInputProps = { ...customProps, ...inputAttributes };
```

Comment:
I'm not spreading the custom props & input attributes on each input will behave as expected.

Reply:
This was carried forward from the previous implementation but I do understand the concern since the same attributes are now being passed to 3 different inputs.
Also a quick reference to `time-input` i see that in a similar way `inputAttributes` are being spread to the different input elements.
@emilyrohrbough, @benbcai Any suggestions as to how to handle this?

Reply:
`inputAttributes` is the first-class prop that comes from DatePicker. The other custom props come from [here](https://github.com/cerner/terra-framework/blob/main/packages/terra-date-picker/src/react-datepicker/index.jsx#L809-L825). Both `inputAttributes` and `customProps` should be spread onto the inputs. If we want to separate these, we would have to create first-class props all the ones we don't already have [here](https://github.com/cerner/terra-framework/blob/main/packages/terra-date-picker/src/react-datepicker/index.jsx#L809-L825).

I don't know that we need to create separate input attribute props for each of the 3 inputs. We can add them when there is a request for it. TimeInput separates these out but I don't see any consumer using them.

Reply:
Unresolving this conversation to talk about it a little more. Spreading inputAttributes onto three different input elements might have some issues in different situations. An `id` for example would spread an id onto each input causing accessibility violations. `disabled` will disable all inputs and buttons holistically, which may or may not be desirable. There is no way to apply a class to an individual input.

What attributes are most commonly being used for `inputAttributes` today and does it make sense to apply them to every input when provided?

Reply:
I agree with @StephenEsser here, I was just trying to resolve issues in `time-input` tests that was a result of the `inputAttributes` being applied to all three inputs, it created duplicate id's on the page, that is not a pattern that I think we should perpetuate.